### ISSUE TITLE
Mask RCNN using CPU instead of GPU for RTX3090

### ISSUE THREAD

**Issue by dyclone** [matterport/Mask_RCNN#2503]
Hello, I've recently switched computers to be able to run the Resnet101 part of the Mask RCNN code. The problem I've been facing is that the program keeps using the CPU instead of the GPU. 

**Current Computer Setup**: 
- NVIDIA RTX 3090 24GB GPU
- AMD Ryzen Threadripper 3960X Processor
- Windows Pro operating system
- CUDA 11.1/ 8.1 cudnn

**What I have tried**:
- I have tried installing CUDA 11.0, cudnn 8.0.4 and upgrading to tensorflow-gpu 1.15. The code runs but since it can’t open files from CUDA 10 it runs on CPU instead. This takes hours to finish running
- There was supposedly an updated version of the code ( leekunhee/Mask_RCNN: Mask R-CNN modified to run on TensorFlow 2 (github.com)). Another github issue tab mentioned using tensorflow 2.3.0, keras 2.4.3, cuda 11.1 cudnn 8.1. I installed everything as instructed but the same thing happens where its asking for CUDA 10 files. I took the older dll CUDA 10 files and put them into the CUDA 11.1 and CUDA 11 bin folder but the program keeps asking for files and eventually asks for files that don’t exist in the CUDA 10 folder

I've looked at plenty of help pages and issues others have posted in this github but haven't been able to find a solution that works. All I been trying to do is find a solution that works with the RTX3090 so I can use the GPU. Any advice would be greatly appreciated. 

Thank you for your time.

**Comment by konstantin-frolov** [comment]
Hi, i have same issue. [This solution](https://github.com/matterport/Mask_RCNN/issues/2427) works for me.

**Comment by dyclone** [comment]
> Hi, i have same issue. [This solution](https://github.com/matterport/Mask_RCNN/issues/2427) works for me.

I've tried that solution but the issue is that the program keeps asking for CUDA 10 dll files. I take the dll files it asks for from the old CUDA 10 bin folder and place them into the CUDA 11 bin folder. Eventually it asks for cupti.dll file. That file doesn't exist in the old CUDA 10 folder. 

Did you also have this issue when using that solution? I also ran the code python setup.py install after setting up the packages. Was I not suppose to do that for the tensorflow 2.0 version of the code?

**Comment by konstantin-frolov** [comment]
> Did you also have this issue when using that solution? I also ran the code python setup.py install after setting up the packages. Was I not suppose to do that for the tensorflow 2.0 version of the code?

I had the same problem. I am using original files from CUDA 11 and Cudnn 8, and delete another versions. It includes the Cupti library.
Try reading this [thread](https://www.reddit.com/r/tensorflow/comments/jsalkw/rtx_3090_and_tensorflow_for_windows_10_step_by/). 
Also if you have any problem with starting cuda, write logs about CUDA initialization from command line here. For example:
`
2021-03-17 16:49:26.764311: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudart64_110.dll
2021-03-17 16:49:26.766807: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublas64_11.dll
2021-03-17 16:49:26.769985: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cublasLt64_11.dll
2021-03-17 16:49:26.772484: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cufft64_10.dll
2021-03-17 16:49:26.774240: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library curand64_10.dll
2021-03-17 16:49:26.775964: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusolver64_11.dll
2021-03-17 16:49:26.778240: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cusparse64_11.dll
2021-03-17 16:49:26.780158: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library cudnn64_8.dll
2021-03-17 16:49:26.781969: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1862] Adding visible gpu devices: 0
2021-03-17 16:49:27.238872: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:
2021-03-17 16:49:27.240678: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      0
2021-03-17 16:49:27.241767: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1280] 0:   N
2021-03-17 16:49:27.242968: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1406] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 19114 MB memory) -> physical GPU (device: 0, name: GeForce RTX 3090, pci bus id: 0000:08:00.0, compute capability: 8.6)`

**Comment by raquelguilhon** [comment]
Hello!

I had the same problem. I was using tensorflow 1.15.0 because I had other issues not solved with TF>2 and MaskRCNN from here.
I noticed that the Build of tensorflow package installed by default when I did _conda install tensorflow=1.15.0_, was _mkl_py36h4920b83_0._
With _conda list tensorflow_, I could see there was a Build called: _**gpu_py**36h4920b83_0_.
That called my attention, then I updated TF:

_**conda install tensorflow=1.15.0=gpu_py36h4920b83_0**_

And that solved my problem. My training is running on GPUs.

CentOS 7
Keras 2.3.1
Tensorflow 1.15.0

**Comment by Behnam72** [comment]
> Hello!
> 
> I had the same problem. I was using tensorflow 1.15.0 because I had other issues not solved with TF>2 and MaskRCNN from here. I noticed that the Build of tensorflow package installed by default when I did _conda install tensorflow=1.15.0_, was _mkl_py36h4920b83_0._ With _conda list tensorflow_, I could see there was a Build called: _**gpu_py**36h4920b83_0_. That called my attention, then I updated TF:
> 
> _**conda install tensorflow=1.15.0=gpu_py36h4920b83_0**_
> 
> And that solved my problem. My training is running on GPUs.
> 
> CentOS 7 Keras 2.3.1 Tensorflow 1.15.0

Can I ask what is your GPU?

**Comment by raquelguilhon** [comment]
> > Hello!
> > I had the same problem. I was using tensorflow 1.15.0 because I had other issues not solved with TF>2 and MaskRCNN from here. I noticed that the Build of tensorflow package installed by default when I did _conda install tensorflow=1.15.0_, was _mkl_py36h4920b83_0._ With _conda list tensorflow_, I could see there was a Build called: _**gpu_py**36h4920b83_0_. That called my attention, then I updated TF:
> > _**conda install tensorflow=1.15.0=gpu_py36h4920b83_0**_
> > And that solved my problem. My training is running on GPUs.
> > CentOS 7 Keras 2.3.1 Tensorflow 1.15.0
> 
> Can I ask what is your GPU?

Sure @Behnam72.

description: 3D controller
product: GV100GL [Tesla V100 SXM2 32GB] [10DE:1DB5]
vendor: NVIDIA Corporation [10DE]
width: 64 bits
clock: 33MHz

CUDA Version: 11.2
